moztelemetry/stats.py

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env python
+# encoding: utf-8
+
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from __future__ import division
+
+import math
+from collections import namedtuple
+
+
+def _rank(sample):
+    """
+    Assign numeric ranks to all values in the sample.
+
+    The ranks begin with 1 for the smallest value. When there are groups of
+    tied values, assign a rank equal to the midpoint of unadjusted rankings.
+
+    E.g.::
+
+        >>> rank({3: 1, 5: 4, 9: 1})
+        {3: 1.0, 5: 3.5, 9: 6.0}
+
+    """
+    rank = 1
+    ranks = {}
+
+    for k in sorted(sample.keys()):
+        n = sample[k]
+        ranks[k] = rank + (n - 1) / 2
+        rank += n
+
+    return ranks
+
+
+def _tie_correct(sample):
+    """
+    Returns the tie correction value for U.
+
+    See: https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.tiecorrect.html
+
+    """
+    tc = 0
+    n = sum(sample.values())
+
+    if n < 2:
+        return 1.0  # Avoid a ``ZeroDivisionError``.
+
+    for k in sorted(sample.keys()):
+        tc += math.pow(sample[k], 3) - sample[k]
+    tc = 1 - tc / (math.pow(n, 3) - n)
+
+    return tc
+
+
+def ndtr(a):
+    """
+    Returns the area under the Gaussian probability density function,
+    integrated from minus infinity to x.
+
+    See: https://docs.scipy.org/doc/scipy/reference/generated/scipy.special.ndtr.html#scipy.special.ndtr
+
+    """
+    try:
+        a = float(a)
```

Comment:
It is probably more pythonic to let this exception be raised. Perhaps just line 72 should be `float(a) * sqrth`. Alternatively, we could just remove validation entirely.